### PR TITLE
Update kafka config

### PR DESCRIPTION
**Pull request by pingsutw**
Signed-off-by: Kevin Su <pingsutw@apache.org>

# TL;DR
Json decoder failed to decode the version in yaml file because the `version` in [sarama.KafkaVersion](https://github.com/Shopify/sarama/blob/978aadd67322a6ee5b790d700698a5ec98aa9170/utils.go#L114) isn't exported. 

Json decoder can only assign the value to the exported field. https://pkg.go.dev/encoding/json#Unmarshal

> Struct values encode as JSON objects. Each exported struct field becomes a member of the object, using the field name as the object key, unless the field is omitted for one of the reasons given below.

- Add `KafkaVersion` in the Kafka config, and use `sarama.ParseKafkaVersion` to parse it.
- Remove `sender.Close` when creating Kafka Client, which is unnecessary. It will be used when sending the event to Kafka.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
^^^

## Tracking Issue
- https://github.com/flyteorg/flyte/issues/3117
- https://flyte-org.slack.com/archives/CP2HDHKE1/p1669385925148139

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by pingsutw** [comment]
Make sense, Updated it, thanks.

File: pkg/runtime/interfaces/application_configuration.go [hunk 1]
```diff
@@ -194,7 +193,8 @@ type GCPConfig struct {
 }
 
 type KafkaConfig struct {
-	Version sarama.KafkaVersion
+	// The version of Kafka, e.g. 2.1.0, 0.8.2.0
+	Version string `json:"kafkaVersion"`
```
Comment (hamersaw) [inline pkg/runtime/interfaces/application_configuration.go:197]:
```suggestion
	Version string `json:"version"`
```

**Review by hamersaw — APPROVED** [review]

**Review by katrogan — APPROVED** [review]
